src/internal/kopia/base_finder.go

```diff
@@ -0,0 +1,259 @@
+package kopia
+
+import (
+	"context"
+	"errors"
+	"sort"
+
+	"github.com/alcionai/clues"
+	"github.com/kopia/kopia/repo/manifest"
+	"github.com/kopia/kopia/snapshot"
+	"golang.org/x/exp/maps"
+
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/internal/model"
+	"github.com/alcionai/corso/src/internal/operations/inject"
+	"github.com/alcionai/corso/src/pkg/backup"
+	"github.com/alcionai/corso/src/pkg/logger"
+)
+
+type BackupBases struct {
+	Backups     []BackupEntry
+	MergeBases  []ManifestEntry
+	AssistBases []ManifestEntry
+}
+
+type BackupEntry struct {
+	*backup.Backup
+	Reasons []Reason
```

Comment:
long term, it seems sensible to add `Reasons` to the backup model.  Thoughts?

Reply:
we'll probably end up with it in the backup _somehow_ though I'm not sure if we should just add it as tags (similar to how it's in snapshots now), as discrete fields, or both